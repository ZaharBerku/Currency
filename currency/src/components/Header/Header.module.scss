@import  "../../style/mixins.scss";
@import "../../style/variables.scss";
@import "../../style/function.scss";


.header{
    &__nav{
        @include flex-container;
        justify-content: space-between;
        padding: 0px adaptiv-indentation(10, 20, 320);
        background: #FFFFFF;
        box-shadow: 0px 6px 10px rgba(229, 229, 229, 0.44);
    };
    .nav{
        &__title{
            @include flex-container;
            gap:24px;
        };
        &__title_svg{
            margin-top: -5px;
            cursor: pointer;
        };
        &__title_text{
            font-weight: 400;
            @include adaptiv-font(42, 30);
            line-height: 61px;
            color:$secondery-color;
            cursor: pointer;
        };
        &__list{
            @include flex-container;
            gap:37px;
            li{
                font-weight: 400;
                font-size: 16px;
                line-height: 23px;
                color:$secondery-color;
            };
        };
        &__cuurrency{
            @include flex-container;
            gap:46px;
            p{
                @include flex-container;
                font-weight: 400;
                font-size: 16px;
                line-height: 23px;
                color:$secondery-color;
                padding-bottom: 9px;
                border-bottom: 1px solid $primary-color;
            };
        };
        &__container_btn{
            border-left: 1px solid $primary-color;
            padding: 15px 0px 15px 46px;
        };
        &__btn_list{
            display: none;    
        };
        &__burgger_line{
            display: none;
        };
        &__btn{
            background: $primary-color;
            border-radius: 21px;
            font-weight: 700;
            font-size: 16px;
            line-height: 23px;
            color: #fff;
            padding: 13px 66px;
            border: none;
            &:hover{
                background: #39118a;
            };
            &:active{
                background: #290c63;
                box-shadow: inset 0px 0px 5px
            };
        };
    };
    @include phone{
        .nav{
            position: relative;
            &__title{
                @include flex-container;
                gap:24px;
            };
            &__list{
                @include flex-container;
                flex-direction: column;
                z-index: 2;
                width: 100vw;
                height: 100vh;
                position: absolute;
                background-color: #fff;
                padding: 0px;
                left: 0px;
                top: 0px;
                order: 3;
                gap:37px;
                display: none;
                li{
                    font-weight: 400;
                    font-size: 16px;
                    line-height: 23px;
                    color:$secondery-color;
                };
            };
            &__cuurrency{
                display: none;
            };
            &__container_btn{
                border-left: 1px solid transparent;
            }
            &__btn_list{
                background: $primary-color;
                border-radius: 21px;
                font-weight: 700;
                font-size: 16px;
                line-height: 23px;
                color: #fff;
                padding: 13px 66px;
                border: none;
                display: block;
                &:hover{
                    background: #39118a;
                };
                &:active{
                    background: #290c63;
                    box-shadow: inset 0px 0px 5px
                };
            };
            &__btn{
              font-size: 0px;
              background: #fff;
              border-radius: 2px;
              padding: 0px;
              width: 25px;
              height: 25px;
              border: none;
              position: relative;
              z-index: 5;
              &::before,
              &::after,
              span{
                content: '';
                transition: all .2s ease;
                position: absolute;
                display: block;
                height: 2px;
                border-radius: 2px;
                width: 100%;
                background: $secondery-color;
                };
                &:hover{
                    background: #fff;
                };
                &:active{
                    background: #fff;
                    box-shadow: inset 0px 0px 0px;
                };
              
              &::before{
                top: 0px;
              };
              &::after{
                top: 8px;
              };
              span{
                top: 16px;
                
              };
            };
            &__open_burger_btn{
                &::before{
                    top: 0px;
                    transform: rotate(45deg);
                  };
                  &::after{
                    top: 0px;
                    transform: rotate(-45deg);
                  };
                  span{
                    top: 16px;
                    transform: scale(0);
                  };
            };
            &__open_burger{
                display: flex;
            };
        };
    };
    @include tablet{
        .nav{
            position: relative;
            &__title{
                @include flex-container;
                gap:24px;
            };
            &__list{
                @include flex-container;
                flex-direction: column;
                z-index: 2;
                width: 100vw;
                height: 100vh;
                position: absolute;
                background-color: #fff;
                padding: 0px;
                left: 0px;
                top: 0px;
                order: 3;
                gap:37px;
                display: none;
                li{
                    font-weight: 400;
                    font-size: 16px;
                    line-height: 23px;
                    color:$secondery-color;
                };
            };
            &__cuurrency{
                display: flex;
            };
            &__container_btn{
                border-left: 1px solid transparent;
            }
            &__btn_list{
                background: $primary-color;
                border-radius: 21px;
                font-weight: 700;
                font-size: 16px;
                line-height: 23px;
                color: #fff;
                padding: 13px 66px;
                border: none;
                display: block;
                &:hover{
                    background: #39118a;
                };
                &:active{
                    background: #290c63;
                    box-shadow: inset 0px 0px 5px
                };
            };
            &__btn{
              font-size: 0px;
              background: #fff;
              border-radius: 2px;
              padding: 0px;
              width: 25px;
              height: 25px;
              border: none;
              position: relative;
              z-index: 5;
              &::before,
              &::after,
              span{
                content: '';
                transition: all .2s ease;
                position: absolute;
                display: block;
                height: 2px;
                border-radius: 2px;
                width: 100%;
                background: $secondery-color;
                };
                &:hover{
                    background: #fff;
                };
                &:active{
                    background: #fff;
                    box-shadow: inset 0px 0px 0px;
                };
              
              &::before{
                top: 0px;
              };
              &::after{
                top: 8px;
              };
              span{
                top: 16px;
                
              };
            };
            &__open_burger_btn{
                &::before{
                    top: 0px;
                    transform: rotate(45deg);
                  };
                  &::after{
                    top: 0px;
                    transform: rotate(-45deg);
                  };
                  span{
                    top: 16px;
                    transform: scale(0);
                  };
            };
            &__open_burger{
                display: flex;
            };
        };
    };
};
