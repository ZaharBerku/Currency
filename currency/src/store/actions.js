import * as actions from "./storeComponents/actions";

export default actions;