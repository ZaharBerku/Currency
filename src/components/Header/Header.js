import React, {useState} from "react";
import { useSelector } from "react-redux";
import classNames from "classnames";
import styles from "./Header.module.scss";
import { GlobalSvg } from "../../assets";
import { List } from "../List";
import { Button } from "../Button";
import { findCcy } from "../../util";

const Header = () => {
    const [isDone, setIsDone] = useState(false);
    const { header } = useSelector(state => state.header);
    const { data } = useSelector(state => state.currency);
    const {nav = [], nameSvg = "", btn = "", USD, EUR } = header;
    
    const dataUSD = findCcy(data, USD);
    const dataEUR = findCcy(data, EUR);

    return (
        <header className={styles.header}>
            <div className={classNames(styles.headerNav, styles.nav)}>
                <div className={styles.navTitle}>
                    <span className={styles.navTitleSvg}>
                    <GlobalSvg type={nameSvg} />
                    </span>
                    <h1 className={styles.navTitleText}>B-bank</h1>
                </div>
                <ul className={classNames(styles.navList, {[styles.navOpenBurger]:isDone})}>
                    {nav.map(({id, text}) =><List key={id} text={text}/>)}
                    <Button additionalClassNames={styles.navBtnList} text={btn.text}/>
                </ul>
                <div className={styles.navCuurrency}>
                {dataUSD && <p><GlobalSvg type={USD} />{`${dataUSD.buy}/${dataUSD.sale}`}</p>}
                {dataEUR && <p><GlobalSvg type={EUR} />{`${dataEUR.buy}/${dataEUR.sale}`}</p>}
                </div>
                <div className={styles.navContainerBtn}>
                <Button additionalClassNames={classNames(styles.navBtn, {[styles.navOpenBurgerBtn]:isDone})} handelClick={()=>setIsDone(!isDone)} >
                    {btn.text}
                    <span className={styles.navBurggerLine}></span>
                </Button>
                </div>
            </div>
        </header>
    );
};

export { Header };