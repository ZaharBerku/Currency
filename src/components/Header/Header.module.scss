@import  "../../style/mixins.scss";
@import "../../style/variables.scss";
@import "../../style/function.scss";


.header{
    &Nav{
        @include flex-container;
        justify-content: space-between;
        padding: 0px adaptiv-indentation(10, 20, 320);
        background: #FFFFFF;
        box-shadow: 0px 6px 10px rgba(229, 229, 229, 0.44);
    };
    .nav{
        &Title{
            @include flex-container;
            gap:24px;
        };
        &TitleSvg{
            margin-top: -5px;
            cursor: pointer;
        };
        &TitleText{
            font-weight: 400;
            @include adaptiv-font(42, 30);
            line-height: 61px;
            color:$secondery-color;
            cursor: pointer;
        };
        &List{
            @include flex-container;
            gap:37px;
            li{
                font-weight: 400;
                font-size: 16px;
                line-height: 23px;
                color:$secondery-color;
            };
        };
        &Cuurrency{
            @include flex-container;
            gap:46px;
            p{
                @include flex-container;
                font-weight: 400;
                font-size: 16px;
                line-height: 23px;
                color:$secondery-color;
                padding-bottom: 9px;
                border-bottom: 1px solid $primary-color;
            };
        };
        &ContainerBtn{
            border-left: 1px solid $primary-color;
            padding: 15px 0px 15px 46px;
        };
        &BtnList{
            display: none;    
        };
        &BurggerLine{
            display: none;
        };
        &Btn{
            background: $primary-color;
            border-radius: 21px;
            font-weight: 700;
            font-size: 16px;
            line-height: 23px;
            color: #fff;
            padding: 13px 66px;
            border: none;
            &:hover{
                background: #39118a;
            };
            &:active{
                background: #290c63;
                box-shadow: inset 0px 0px 5px
            };
        };
    };
    @include phone{
        .nav{
            position: relative;
            &Title{
                @include flex-container;
                gap:24px;
            };
            &List{
                @include flex-container;
                flex-direction: column;
                z-index: 2;
                width: 100vw;
                height: 100vh;
                position: absolute;
                background-color: #fff;
                padding: 0px;
                left: 0px;
                top: 0px;
                order: 3;
                gap:37px;
                display: none;
                li{
                    font-weight: 400;
                    font-size: 16px;
                    line-height: 23px;
                    color:$secondery-color;
                };
            };
            &Cuurrency{
                display: none;
            };
            &ContainetBtn{
                border-left: 1px solid transparent;
            }
            &BtnList{
                background: $primary-color;
                border-radius: 21px;
                font-weight: 700;
                font-size: 16px;
                line-height: 23px;
                color: #fff;
                padding: 13px 66px;
                border: none;
                display: block;
                &:hover{
                    background: #39118a;
                };
                &:active{
                    background: #290c63;
                    box-shadow: inset 0px 0px 5px
                };
            };
            &Btn{
              font-size: 0px;
              background: #fff;
              border-radius: 2px;
              padding: 0px;
              width: 25px;
              height: 25px;
              border: none;
              position: relative;
              z-index: 5;
              &::before,
              &::after,
              span{
                content: '';
                transition: all .2s ease;
                position: absolute;
                display: block;
                height: 2px;
                border-radius: 2px;
                width: 100%;
                background: $secondery-color;
                };
                &:hover{
                    background: #fff;
                };
                &:active{
                    background: #fff;
                    box-shadow: inset 0px 0px 0px;
                };
              
              &::before{
                top: 0px;
              };
              &::after{
                top: 8px;
              };
              span{
                top: 16px;
                
              };
            };
            &OpenBurgerBtn{
                &::before{
                    top: 0px;
                    transform: rotate(45deg);
                  };
                  &::after{
                    top: 0px;
                    transform: rotate(-45deg);
                  };
                  span{
                    top: 16px;
                    transform: scale(0);
                  };
            };
            &OpenBurger{
                display: flex;
            };
        };
    };
    @include tablet{
        .nav{
            position: relative;
            &Title{
                @include flex-container;
                gap:24px;
            };
            &List{
                @include flex-container;
                flex-direction: column;
                z-index: 2;
                width: 100vw;
                height: 100vh;
                position: absolute;
                background-color: #fff;
                padding: 0px;
                left: 0px;
                top: 0px;
                order: 3;
                gap:37px;
                display: none;
                li{
                    font-weight: 400;
                    font-size: 16px;
                    line-height: 23px;
                    color:$secondery-color;
                };
            };
            &Cuurrency{
                display: flex;
            };
            &ContainerBtn{
                border-left: 1px solid transparent;
            }
            &BtnList{
                background: $primary-color;
                border-radius: 21px;
                font-weight: 700;
                font-size: 16px;
                line-height: 23px;
                color: #fff;
                padding: 13px 66px;
                border: none;
                display: block;
                &:hover{
                    background: #39118a;
                };
                &:active{
                    background: #290c63;
                    box-shadow: inset 0px 0px 5px
                };
            };
            &Btn{
              font-size: 0px;
              background: #fff;
              border-radius: 2px;
              padding: 0px;
              width: 25px;
              height: 25px;
              border: none;
              position: relative;
              z-index: 5;
              &::before,
              &::after,
              span{
                content: '';
                transition: all .2s ease;
                position: absolute;
                display: block;
                height: 2px;
                border-radius: 2px;
                width: 100%;
                background: $secondery-color;
                };
                &:hover{
                    background: #fff;
                };
                &:active{
                    background: #fff;
                    box-shadow: inset 0px 0px 0px;
                };
              
              &::before{
                top: 0px;
              };
              &::after{
                top: 8px;
              };
              span{
                top: 16px;
                
              };
            };
            &OpenBurgerBtn{
                &::before{
                    top: 0px;
                    transform: rotate(45deg);
                  };
                  &::after{
                    top: 0px;
                    transform: rotate(-45deg);
                  };
                  span{
                    top: 16px;
                    transform: scale(0);
                  };
            };
            &OpenBurger{
                display: flex;
            };
        };
    };
};
