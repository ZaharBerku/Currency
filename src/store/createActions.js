export {
      //SELECT
  getOptions,
  filterOptions,
  reverseValueSelect,
  getCcy,
  getBaseCcy,
  
  //INPUT
  changeInputValue,
  getValueInput,
  isDisableInput,
  reverseValueInputs,

  //CURRENCY
  getDataCurrency,

  //HEADER 
  getDataHeader
} from "./storeComponents/createActions";

