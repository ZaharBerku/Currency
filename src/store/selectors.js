import * as selectors from "./storeComponents/select";

export { selectors };