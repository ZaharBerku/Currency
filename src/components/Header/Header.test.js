/* eslint-disable no-undef */
import React from "react";
import { Header } from "./Header";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../store/store";

const Component = () => {
    return(
        <Provider store={store}>
            <Header/>
        </Provider>
    );
};

describe("Header snapshot render", () => {
    test("should Header render", () => {
        const { asFragment } = render(<Component/>);
        expect(asFragment()).toMatchSnapshot();
    });
});