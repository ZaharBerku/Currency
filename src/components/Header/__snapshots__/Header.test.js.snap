// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot render should Header render 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="headerNav nav"
    >
      <div
        class="navTitle"
      >
        <span
          class="navTitleSvg"
        />
        <h1
          class="navTitleText"
        >
          B-bank
        </h1>
      </div>
      <ul
        class="navList"
      >
        <button
          class="navBtnList"
          data-testid="button"
        />
      </ul>
      <div
        class="navCuurrency"
      />
      <div
        class="navContainerBtn"
      >
        <button
          class="navBtn"
          data-testid="button"
        >
          <span
            class="navBurggerLine"
          />
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
